aiomcache/pool.py

```diff
@@ -39,29 +37,20 @@ def acquire(self):
 
         :return: ``tuple`` (reader, writer)
         """
-        while self._size < self._minsize:
+        while self.size() < self._minsize:
             _conn = yield from self._create_new_conn()
-
-            # Could not create new connection
             if _conn is None:
                 break
-            yield from self._pool.put(_conn)
+            self._pool.put_nowait(_conn)
 
         conn = None
         while not conn:
-            if not self._pool.empty():
-                _conn = yield from self._pool.get()
-                if _conn.reader.at_eof() or _conn.reader.exception():
-                    self._do_close(_conn)
-                    conn = None
-                else:
-                    conn = _conn
-
-            if conn is None:
+            _conn = yield from self._pool.get()
```

Comment:
Clients will now block here until there is a connection available. If the retrieved connection has an error, they will close it and try to create a new one